data/transactions/logic/assembler.go

```diff
@@ -1591,41 +1629,45 @@ func joinIntsOnOr(singularTerminator string, list ...int) string {
 	return errMsg + singularTerminator + "s"
 }
 
-func pseudoImmediatesError(ops *OpStream, name string, specs map[int]OpSpec) {
+func pseudoImmediatesError(ops *OpStream, mnemonic token, specs map[int]OpSpec) {
 	immediateCounts := make([]int, len(specs))
 	i := 0
 	for numImms := range specs {
 		immediateCounts[i] = numImms
 		i++
 	}
-	ops.error(name + " expects " + joinIntsOnOr("immediate argument", immediateCounts...))
+	ops.record(mnemonic.errorf("%s expects %s", mnemonic.str, joinIntsOnOr("immediate argument", immediateCounts...)))
 }
 
 // getSpec finds the OpSpec we need during assembly based on its name, our current version, and the immediates passed in
 // Note getSpec handles both normal OpSpecs and those supplied by versionedPseudoOps
 // The returned string is the spec's name, annotated if it was a pseudoOp with no immediates to help disambiguate typetracking errors
-func getSpec(ops *OpStream, name string, args []string) (OpSpec, string, bool) {
+func getSpec(ops *OpStream, mnemonic token, args int) (OpSpec, string, bool) {
```

Comment:
nit: consider renaming `args` to `argNum`
```suggestion
func getSpec(ops *OpStream, mnemonic token, argNum int) (OpSpec, string, bool) {
```